src/components/ContentHighlights/HighlightStepper/HighlightStepperSelectContentDataTable.jsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { useContextSelector } from 'use-context-selector';
+import { Configure, InstantSearch, connectStateResults } from 'react-instantsearch-dom';
+import { DataTable, CardView } from '@edx/paragon';
+import { camelCaseObject } from '@edx/frontend-platform';
+
+import { configuration } from '../../../config';
+import { FOOTER_TEXT_BY_CONTENT_TYPE } from '../data/constants';
+import ContentSearchResultCard from './ContentSearchResultCard';
+import { ContentHighlightsContext } from '../ContentHighlightsContext';
+import SelectContentSelectionStatus from './SelectContentSelectionStatus';
+import SelectContentSelectionCheckbox from './SelectContentSelectionCheckbox';
+import SelectContentSearchPagination from './SelectContentSearchPagination';
+import SkeletonContentCard from '../SkeletonContentCard';
+import { useContentHighlightsContext } from '../data/hooks';
+
+const defaultActiveStateValue = 'card';
+const pageSize = 24;
+
+const selectColumn = {
+  id: 'selection',
+  Header: () => null,
+  Cell: SelectContentSelectionCheckbox,
+  disableSortBy: true,
+};
+
+const prodEnterpriseId = 'e783bb19-277f-479e-9c41-8b0ed31b4060';
+const currentEpoch = Math.round((new Date()).getTime() / 1000);
+
+const HighlightStepperSelectContent = () => {
+  const { setCurrentSelectedRowIds } = useContentHighlightsContext();
+  const currentSelectedRowIds = useContextSelector(
+    ContentHighlightsContext,
+    v => v[0].stepperModal.currentSelectedRowIds,
+  );
+  const searchClient = useContextSelector(
+    ContentHighlightsContext,
+    v => v[0].searchClient,
+  );
+
+  const searchFilters = `enterprise_customer_uuids:${prodEnterpriseId} AND advertised_course_run.upgrade_deadline > ${currentEpoch} AND content_type:course`;
+  return (
+    <InstantSearch
+      indexName={configuration.ALGOLIA.INDEX_NAME}
+      searchClient={searchClient}
+    >
+      <Configure
+        filters={searchFilters}
+        hitsPerPage={pageSize}
+      />
+      <HighlightStepperSelectContentDataTable
+        selectedRowIds={currentSelectedRowIds}
+        onSelectedRowsChanged={setCurrentSelectedRowIds}
+      />
+    </InstantSearch>
+  );
+};
+
+const PriceTableCell = ({ row }) => {
+  const contentPrice = row.original.firstEnrollablePaidSeatPrice;
```

Comment:
[note] tests will be added for covered in a future PR

Reply:
Leaving open as a reminder to complete